core/src/main/java/com/sequenceiq/cloudbreak/service/RotateSaltPasswordService.java

```diff
@@ -95,4 +124,40 @@ private void sendUsageReport(String resourceCrn, RotateSaltPasswordReason reason
                     resourceCrn, reason, result, message, e);
         }
     }
+
+    public Optional<RotateSaltPasswordReason> checkIfSaltPasswordRotationNeeded(StackDto stack) {
+        Optional<RotateSaltPasswordReason> result;
```

Comment:
could you initialize this with empty, as optional must not return with null?

Reply:
It is not null (it is uninitialized), so IDEA will scream if you leave it that way and try to return it

Reply:
okay, I know, it's for future proofing